File: mobile-ui-testing-android/app/src/androidTest/java/biz/tearoom6/mobile_ui_testing_android/uitest/helper/NoteHelper.java
package biz.tearoom6.mobile_ui_testing_android.uitest.helper;

import biz.tearoom6.mobile_ui_testing_android.R;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.click;
import static android.support.test.espresso.action.ViewActions.replaceText;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static android.support.test.espresso.matcher.ViewMatchers.withText;
import static biz.tearoom6.mobile_ui_testing_android.uitest.helper.ViewHelper.selectTab;
import static org.apache.commons.lang3.RandomStringUtils.random;

public class NoteHelper {

    public static String randomTitle() {
        return "Title:" + random(5, 'あ', 'ん', false, false) + System.currentTimeMillis();
    }

    public static String randomContent() {
        return "Content:" + random(5, 'あ', 'ん', false, false) + System.currentTimeMillis();
    }

    public static void createNote(String title, String content) {
        selectTab(withText(R.string.tab_note));

        // Click create button.
        onView(withId(R.id.notes_button_create)).perform(click());

        // Input title & content.
        onView(withId(R.id.note_input_title)).perform(replaceText(title));
        onView(withId(R.id.note_input_content)).perform(replaceText(content));

        // Click save button.
        onView(withId(R.id.note_button_save)).perform(click());
    }
}

File: mobile-ui-testing-appium/spec/support/helpers/note_helper.rb
require 'randexp-multibyte'

module NoteHelper
  include ElementHelper

  def random_title
    "Title: #{/[:japanese:]{10}/.gen}"
  end

  def random_content
    "Content: #{/[:japanese:]{10}/.gen}"
  end

  def create_note(title:, content:)
    click_note_tab

    # Click create button.
    click_button(id: 'notes_button_create')
    click_button(id: 'note_button_clear')

    # Input title & content.
    input_text(id: 'note_input_title', text: title)
    input_text(id: 'note_input_content', text: content)

    # Click save button.
    click_button(id: 'note_button_save')
  end

  def note_titles
    click_note_tab

    note_titles_list
  end

  def latest_note_title
    note_titles.first.text
  end

  def click_note_tab
    click_tab(name: 'Note')
  end
end
